hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordMerger.java

```diff
@@ -43,9 +44,27 @@ public interface HoodieRecordMerger extends Serializable {
    * This method converges combineAndGetUpdateValue and precombine from HoodiePayload.
    * It'd be associative operation: f(a, f(b, c)) = f(f(a, b), c) (which we can translate as having 3 versions A, B, C
    * of the single record, both orders of operations applications have to yield the same result)
+   * This method takes only full records for merging.
    */
   Option<Pair<HoodieRecord, Schema>> merge(HoodieRecord older, Schema oldSchema, HoodieRecord newer, Schema newSchema, TypedProperties props) throws IOException;
 
+  /**
+   * Merges records which can contain partial updates, i.e., only subset of fields and values are
+   * present in the record representing the updates, and absent fields are not updated.
```

Comment:
Maybe a simple demo to demonstrate the update behaviors, like what we do in `PartialUpdateAvroPayload`.

Reply:
Examples added in docs.